File: Week 2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

int main (int argc, string argv[])

{
    if (argc!=2)
        {
            printf("Please enter one non-negative number\n");
            return 1;
        }
    printf("plaintext:");
    string plaintext=get_string();
    if (plaintext!=NULL)
{
    int key= atoi(argv[1]);
    int letter=0;
    
    printf("ciphertext:");
    
    for (int i=0, length= strlen(plaintext); i<length;i++)
    {
    
     if (isupper(plaintext[i]))
        {
                char l= plaintext[i];
                letter= (int)l;
                letter-= 65;
                letter= (letter+key)%26;
                letter+= 65;
                printf ("%c", (char)letter);
        }
    else if (islower(plaintext[i]))
        {
                char l= plaintext[i];
                letter= (int)l;
                letter-= 97;
                letter= (letter+key)%26;
                letter+= 97;
                printf ("%c", (char)letter);
        }
    else
        {
            printf("%c", plaintext[i]);
        }
    }

    printf("\n");
}
}

File: Week 8 -no pset for Week 6 and 7/caesar.py
import cs50
import sys

def main():
    #pass in key and plaintext
    if len(sys.argv)!=2:
        print("Please enter one non-negative number")
        exit(1)
    print("plaintext:", end="")
    plaintext=cs50.get_string()
    print("ciphertext:", end="")
    
    #cipher text
    if plaintext!=None:
        key=int(sys.argv[1])
        cipher_text(plaintext, key)


def cipher_text(plaintext, key):
    for i in plaintext:
            
            if 65<=ord(i)<=90:
                letter=ord(i)
                letter-=65
                letter=(letter+key)%26
                letter+=65
                print("{}".format(chr(letter)), end="")
            elif 97<=ord(i)<=122:
                letter=ord(i)
                letter-=97
                letter=(letter+key)%26
                letter+=97
                print("{}".format(chr(letter)), end="")
            else:
                print("{}".format(i), end="")
    print()

if __name__ == "__main__":
    main()
